src/mediarithmics/plugins/common/BasePlugin.ts

```diff
@@ -127,6 +127,8 @@ export abstract class BasePlugin {
   INSTANCE_CONTEXT_CACHE_EXPIRATION: number = 120000;
 
   pluginCache: any;
+  private dataFileCachingDate: string;
+  private dataFileCache: Buffer;
```

Comment:
You can have *a lot* of files pulled by a plugin (here you will mix contents randomly), which can take a lot of memory (with a `Buffer`).

Reply:
@dduponchel > I'm not sure to perfectly understand your point.

I see 2 potential remarks in your comment:
a. The cache is not per URI in this impl. which is wrong
b. We're going to cache all the entire DataFiles in the plugin memory which could be a bad idea if there is too many instances with too many DataFile (which would be too big of course)

Was it both?

On a. > true, see my remark below.
On b. > Yes, but we don't know how to do it without breaking the Plugin SDK API -> `fetchDataFile(...)` is always returning a Buffer in the v0.7.x release. So caching the DataFile buffer seems the only option.

Releasing a new major version to embed the breaking change would do the trick. However, we have big plans for the v0.8.x:
In the v0.8.x, we would like to completely rethink the way we're calling + working with the `instanceContextBuilder()`

Currently, we're periodically calling it, and it fetch the required inputs (datafiles, properties, etc.).

We would like to try to fetch all the necessary inputs from within the SDK and call the `instanceContextBuilder` only when one of them was updated. A bit like how "nix" is working -> we'll also ask the dev. to have a determinitic impl. of the `instanceContextBuilder()`.

That will require a lot of breaking API changes and will need to ask Plugin devs to migrate their code.

If we were to update the `fetchDataFile()` signature in the short term to return optionally a Buffer and stop caching it in the SDK as proposed here, our big plans would be delayed until the v0.9.x as the v0.8.x would be targeting the `fetchDataFile()` signature change.

What do you think? Caching the DataFile buffer in memory is really a no go for you? And we should release 2 major versions for each change (v0.8.x -> `fetchDataFile` / v0.9.x -> `instanceContextBuilder`)?

@jmunozzz > Each time that `fetchDataFile` will be called with a different uri, it will match a different DataFile (and thus another content).

As you have currently a global cache, and not a cache "sharded" by uri, you'll mix up datafiles between them when returning the cache version.

Asking for uri=A could return the "cached" version that could be from a prior call to the function with uri=b (and not the same file)

Reply:
@PoulpiFr I see. I thought there were a single datafile.